src/app/home-page/widget/entry-box/entry-box.component.ts

```diff
@@ -0,0 +1,101 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { EntryType } from 'app/shared/enum/entry-type';
+import { EntryUpdateTime, User, UsersService } from 'app/shared/openapi';
+import { UserQuery } from 'app/shared/user/user.query';
+import { debounceTime, finalize, takeUntil } from 'rxjs/operators';
+import { MyWorkflowsService } from 'app/myworkflows/myworkflows.service';
+import { RegisterToolService } from 'app/container/register-tool/register-tool.service';
+import { Base } from 'app/shared/base';
+import { Dockstore } from 'app/shared/dockstore.model';
+
+@Component({
+  selector: 'app-entry-box',
+  templateUrl: './entry-box.component.html',
+  styleUrls: ['./entry-box.component.scss'],
+})
+export class EntryBoxComponent extends Base implements OnInit {
+  Dockstore = Dockstore;
+  @Input() entryType: EntryUpdateTime.EntryTypeEnum;
+  entryTypeLowerCase: string;
+  filterText: string;
+  listOfEntries: Array<EntryUpdateTime> = [];
+  user: User;
+  helpLink: string;
+  allEntriesLink: string;
+  hasEntries: boolean;
+  firstLoad = true;
+  public isLoading = true;
+
+  constructor(
+    private registerToolService: RegisterToolService,
+    private userQuery: UserQuery,
+    private myWorkflowsService: MyWorkflowsService,
+    private usersService: UsersService
+  ) {
+    super();
+    this.user = this.userQuery.getValue().user;
+  }
+
+  ngOnInit(): void {
+    this.getMyEntries();
+
+    if (this.entryType) {
+      this.entryTypeLowerCase = this.entryType.toLowerCase();
+    }
+
+    //Get the links for the specified entryType
+    if (this.entryType === EntryUpdateTime.EntryTypeEnum.WORKFLOW) {
+      this.helpLink = Dockstore.DOCUMENTATION_URL + '/getting-started/dockstore-workflows.html';
+      this.allEntriesLink = '/my-workflows/';
+    } else if (this.entryType === EntryUpdateTime.EntryTypeEnum.TOOL) {
+      this.helpLink = Dockstore.DOCUMENTATION_URL + '/getting-started/dockstore-tools.html';
+      this.allEntriesLink = '/my-tools/';
+    } else if (this.entryType === EntryUpdateTime.EntryTypeEnum.SERVICE) {
+      this.helpLink = Dockstore.DOCUMENTATION_URL + '/getting-started/getting-started-with-services.html';
+      this.allEntriesLink = '/my-services/';
+    }
+  }
+
+  getMyEntries() {
+    this.usersService
+      .getUserEntries(null, this.filterText)
+      .pipe(
+        finalize(() => (this.isLoading = false)),
+        debounceTime(750),
+        takeUntil(this.ngUnsubscribe)
+      )
+      .subscribe((myEntries: Array<EntryUpdateTime>) => {
+        this.listOfEntries = [];
+        myEntries
+          .filter(
+            (entry) =>
+              entry.entryType === this.entryType ||
+              (this.entryType === EntryUpdateTime.EntryTypeEnum.TOOL && entry.entryType === EntryUpdateTime.EntryTypeEnum.APPTOOL)
```

Comment:
This won't work if `this.entryType` is `EntryType.EntryType.APPTOOL`, but maybe we don't need to worry about it.

Reply:
I don't think `this.entryType` will ever be `APPTOOL` because based off of the three boxes and calling this component, it can only be `workflow` `tool` or `service`, I'm just grouping the APPTOOLs into the same tools box. I think this makes sense?